packages/sdk/src/utils/index.ts

```diff
@@ -0,0 +1,369 @@
+import * as ecc from "@bitcoinerlab/secp256k1";
+import { BIP32Interface } from "bip32";
+import {
+  crypto,
+  initEccLib,
+  Network as BitcoinNetwork,
+  networks,
+  Payment,
+  PaymentCreator,
+  payments,
+  Psbt,
+  Signer,
+  Transaction,
+} from "bitcoinjs-lib";
+import { Buffer } from "buffer";
+import ECPairFactory, { ECPairInterface } from "ecpair";
+
+import { ADDRESS_TYPE_TO_FORMAT } from "../addresses/constants";
+import type { AddressFormat, AddressType } from "../addresses/types";
+import type { Network } from "../config/types";
+import { OrditSDKError } from "../errors";
+import type { UTXO } from "../transactions/types";
+import type {
+  BufferOrHex,
+  EncodeDecodeObjectOptions,
+  GetScriptTypeResponse,
+  IsBitcoinPaymentResponse,
+  NestedObject,
+  OneOfAllDataFormats,
+} from "./types";
+
+export function getNetwork(value: Network) {
+  if (value === "mainnet") {
+    return networks.bitcoin;
+  }
+
+  return networks[value];
+}
+
+export function createTransaction(
+  key: Buffer,
+  type: AddressType,
+  network: Network | BitcoinNetwork,
+  paymentOptions?: Payment,
+) {
+  initEccLib(ecc);
+  const networkObj =
+    typeof network === "string" ? getNetwork(network) : network;
+
+  if (type === "p2tr") {
+    return payments.p2tr({
+      internalPubkey: key,
+      network: networkObj,
+      ...paymentOptions,
+    });
+  }
+
+  if (type === "p2sh") {
+    return payments.p2sh({
+      redeem: payments.p2wpkh({ pubkey: key, network: networkObj }),
+      network: networkObj,
+    });
+  }
+
+  return payments[type]({ pubkey: key, network: networkObj });
+}
+
+export function getDerivationPath(
+  formatType: AddressFormat,
+  account = 0,
+  addressIndex = 0,
+) {
+  const pathFormat: Record<AddressFormat, string> = {
+    legacy: `m/44'/0'/${account}'/0/${addressIndex}`,
+    "p2sh-p2wpkh": `m/49'/0'/${account}'/0/${addressIndex}`,
+    segwit: `m/84'/0'/${account}'/0/${addressIndex}`,
+    taproot: `m/86'/0'/${account}'/0/${addressIndex}`,
+  };
+  return pathFormat[formatType];
+}
+
+export function hdNodeToChild(
+  node: BIP32Interface,
+  formatType: AddressFormat = "legacy",
+  addressIndex = 0,
+  account = 0,
+) {
+  const fullDerivationPath = getDerivationPath(
+    formatType,
+    account,
+    addressIndex,
+  );
+
+  return node.derivePath(fullDerivationPath);
+}
+
+export function toXOnly(pubkey: Buffer): Buffer {
+  return pubkey.subarray(1, 33);
+}
```

Comment:
This is a duplicate fn. IIRC, this already exists in bitcoinjs-lib. Worth double-checking.

Reply:
Not exported by them.. added a comment as well for this function.